### PR TITLE
Fix/spatial temporal

### PR DESCRIPTION
**Pull request by Xiangyu-Hu**
1. revised the free surface detecting algorithm for easier understanding the generalization.
2. revise the 2d dambreak and static confinement case data base due to last fix on wall condition.

### CODE REVIEW

**Review by FabienPean-Virtonomy — APPROVED** [review]

File: tests/2d_examples/test_2d_static_confinement/regression_test_tool/FluidObserver_Pressure_Run_0_result.xml [hunk 1]
```diff
@@ -4,6 +4,10 @@
...
+=======
         <Particle_0 snapshot_0="0" snapshot_1="0" snapshot_2="0" snapshot_3="0.33475922210654263" snapshot_4="1.1333527423978724" snapshot_5="0.81054965104114274" snapshot_6="0.62524361075552337" snapshot_7="1.2832297853467527" snapshot_8="0.21780461066616247" snapshot_9="0.6982746528432846" snapshot_10="0.2289733207042875" snapshot_11="0.3622451915049627" snapshot_12="0.44662296089815118" snapshot_13="0.34021949324082545" snapshot_14="0.32241323280681866" snapshot_15="0.2628323452481528" snapshot_16="0.18970249266384329" snapshot_17="0.10565258502415824" snapshot_18="0.0898196626765419" snapshot_19="0.1323653620948424" snapshot_20="0.11701220076416403" snapshot_21="0.037385998165414353" />
+>>>>>>> static confinement new data set
```
[the remark this answers is not in the document]
Comment (Xiangyu-Hu) [reply tests/2d_examples/test_2d_static_confinement/regression_test_tool/FluidObserver_Pressure_Run_0_result.xml:11]: OK now. May XML reading check the data by key word.

File: tests/2d_examples/test_2d_static_confinement/regression_test_tool/WaterBody_TotalMechanicalEnergy_dtwdistance.xml [hunk 2]
```diff
@@ -1,4 +1,8 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <dtw_distance>
+<<<<<<< HEAD
+    <DTWDistance TotalMechanicalEnergy_0="0.0619131055010545" />
+=======
     <DTWDistance TotalMechanicalEnergy_0="0.1" />
+>>>>>>> static confinement new data set
```
[the remark this answers is not in the document]
Comment (Xiangyu-Hu) [reply tests/2d_examples/test_2d_static_confinement/regression_test_tool/WaterBody_TotalMechanicalEnergy_dtwdistance.xml:7]: same.

File: tests/2d_examples/test_2d_static_confinement/regression_test_tool/WaterBody_TotalMechanicalEnergy_runtimes.dat [hunk 3]
```diff
@@ -1,3 +1,7 @@
 true
+<<<<<<< HEAD
+8
+=======
 14
+>>>>>>> static confinement new data set
```
[the remark this answers is not in the document]
Comment (Xiangyu-Hu) [reply tests/2d_examples/test_2d_static_confinement/regression_test_tool/WaterBody_TotalMechanicalEnergy_runtimes.dat:6]: same.

File: tests/2d_examples/test_2d_static_confinement/regression_test_tool/FluidObserver_Pressure_runtimes.dat [hunk 4]
```diff
@@ -1,3 +1,7 @@
 true
+<<<<<<< HEAD
+10
+=======
 11
+>>>>>>> static confinement new data set
```
[the remark this answers is not in the document]
Comment (Xiangyu-Hu) [reply tests/2d_examples/test_2d_static_confinement/regression_test_tool/FluidObserver_Pressure_runtimes.dat:6]: Same.

**Pull request merged**